gateway-messages/src/sp_impl.rs

```diff
@@ -697,27 +635,24 @@ fn handle_message_impl<H: SpHandler>(
 }
 
 fn handle_mgs_response<H: SpHandler>(
-    sender: SocketAddrV6,
-    port: SpPort,
+    sender: Sender,
     message_id: u32,
     handler: &mut H,
     kind: MgsResponse,
     leftover: &[u8],
 ) {
+    // TODO check sender here?
```

Comment:
I think yes? `faux-mgs` will never send a `MessageKind::MgsResponse`, but an arbitrary program could, and they should presumably be filtered out immediately if they're coming from a locked tech port.